src/contract.rs

```diff
@@ -428,12 +432,9 @@ fn replace_scope_owner(
         address: new_owner.clone(),
         role: PartyType::Owner,
     });
-    // Upon request, also swap over the value owner.  The contract cannot transfer value ownership to itself
-    // during contract execution, so this functionality should only be used when the contract is already the
-    // value owner of a scope and wants to swap it to a different owner
-    if replace_value_owner {
-        scope.value_owner_address = new_owner;
-    }
+    // Swap over the value owner, ensuring that the target owner not only is listed as an owner,
+    // but has full access control over the scope
+    scope.value_owner_address = new_owner;
```

Comment:
This functionality now will always want to change both the owners vector AND the value owner, so providing the option to skip it is a bad idea because it can leave the scope in an orphaned state where it's owned by the contract and no operations can be done to swap the owner after that point